### ISSUE TITLE
Dev Branch 'SocketPool' object has no attribute 'timeout'

### ISSUE THREAD

**Issue by makermelissa** [adafruit/Adafruit_CircuitPython_MiniMQTT#58]
I found this on the ESP32-S2 Metro using the Dev Branch. It only happens occasionally"
```
Traceback (most recent call last):
  File "code.py", line 140, in <module>
  File "/lib/adafruit_minimqtt/adafruit_minimqtt.py", line 862, in loop
  File "/lib/adafruit_minimqtt/adafruit_minimqtt.py", line 874, in _wait_for_msg
AttributeError: 'SocketPool' object has no attribute 'timeout'
```

**Comment by makermelissa** [comment]
I replaced line 874 with `except OSError:` and that seems to be working now.

**Comment by brentru** [comment]
@makermelissa  Thanks for finding this type of bug. Which `OSError` does it throw?

**Comment by makermelissa** [comment]
That's the thing, it never does throw an error. It just didn't like that timeout didn't exist in SocketPool, which is why I noted that it's probably not the right way to do it.

**Comment by makermelissa** [comment]
Ok, removed the try/except bloack and got:
```
OSError: [Errno 116] ETIMEDOUT
```

**Comment by brentru** [comment]
Thanks for finding it, I'll catch that specific error

**Comment by makermelissa** [comment]
Ok this code block seems pretty solid:
```
        try:
            self._sock.recv_into(res, 1)
            print("Resp: ", res) # TODO BR: Remove debugging
        except OSError as error:
            if error.errno == errno.ETIMEDOUT:
                return None
            else:
                raise
```

**Comment by brentru** [comment]
This has been addressed in #61 and released.
